src/components/LinkChooser/types.ts

```diff
@@ -0,0 +1,134 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { CheckboxState } from "@components/Checkbox/Checkbox";
+import { MenuItemType, MenuBlock } from "@components/Menu/SelectMenu";
+import { Validation } from "@components/TextInput/TextInput";
+import { AriaListBoxOptions } from "@react-aria/listbox";
+import { ListState } from "@react-stately/list";
+import { KeyboardEvent, FocusEvent } from "react";
+import { RefObject, ReactNode, HTMLInputTypeAttribute, ReactElement, ChangeEventHandler } from "react";
+import { Interpreter, DoneInvokeEvent } from "xstate";
+import { Node } from "@react-types/shared";
+
+export type MachineService = Interpreter<
+    LinkChooserContext,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    any,
+    DoneInvokeEvent<LinkChooserEventData>,
+    {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        value: any;
+        context: LinkChooserContext;
+    }
+>;
+
+export type TemplateMenuBlock = {
+    id: string;
+    menuItems: SearchResult[];
+    ariaLabel?: string;
+};
+
+export type LinkChooserProps = {
+    openInNewTab: boolean;
+    ariaLabel?: string;
+    label?: string;
+    placeholder?: string;
+    onOpenInNewTabChange: (value: boolean) => void;
+    onLinkChange: (value: SearchResult | null) => void;
+    readonly copyToClipboard: Clipboard;
+    readonly getGlobalByQuery?: (query: string) => Promise<SearchResult[]>;
+    readonly getGuidelinesByQuery?: (query: string) => Promise<SearchResult[]>;
+    readonly getTemplatesByQuery?: (query: string) => Promise<SearchResult[]>;
+    readonly openPreview: (value: string, target: string) => void;
+};
+
+export type SearchResult = Omit<MenuItemType, "title"> & { icon: string; title: string };
+
+export type SearchResultListProps = AriaListBoxOptions<unknown> & {
+    listBoxRef?: RefObject<HTMLUListElement>;
+    state: ListState<unknown>;
+    menuBlocks: (Omit<MenuBlock, "menuItems"> & { menuItems: SearchResult[] })[];
+    noBorder?: boolean;
+    hasItems?: boolean;
+    query: string;
+    machineService: MachineService;
+};
+
+export type SearchResultSectionProps = {
+    heading: Node<unknown>;
+    state: ListState<unknown>;
+    keyItemRecord: Record<string, SearchResult>;
+    machineService: MachineService;
+};
+
+export type SearchResultOptionProps = {
+    item: Node<unknown>;
+    state: ListState<unknown>;
+    keyItemRecord: Record<string, SearchResult>;
+    machineService: MachineService;
+};
+
+export type TemplateProps = {
+    title: ReactNode;
+    subtitle?: string;
+    preview?: string;
+};
+
+export type SectionActionMenuProps = {
+    machineService: MachineService;
+};
+
+export type SearchInputProps = {
```

Comment:
these are maaany props ;) aren't the most of them part of react-aria, so that we can reuse it's types and spread it?

Reply:
this was taken directly from the `TextInput` component props, with added custom properties and methods for state manipulation